### PR TITLE
deferring public ip calculation to a place that doesn't run in a VPC

### PR DESCRIPTION
**Pull request by zmarois**

### CODE REVIEW

File: priam/src/main/java/com/netflix/priam/defaultimpl/PriamConfiguration.java [hunk 1]
```diff
@@ -683,7 +673,7 @@ public boolean isIncrBackup() {
     @Override
     public String getHostIP() {
         if (this.isVpcRing()) return LOCAL_IP;
-        else return PUBLIC_IP;
+        else return getInstanceDataRetriever().getPublicHostname();
```
Comment (arunagrawal84) [inline priam/src/main/java/com/netflix/priam/defaultimpl/PriamConfiguration.java:676]: Why not getInstanceDataRetriever().getPublicIP();

Comment (ryus08) [reply priam/src/main/java/com/netflix/priam/defaultimpl/PriamConfiguration.java]: Good catch, I messed up here.

My primary purpose of doing this is just to defer the call to getPublicHostname/getPublicIP to the time it is actually used. My instances in a private VPC don't have these, so I didn't personally run into an issue.

Would you like this upstream if done correctly? I'd like to search for other usages of instance metadata and isolate into the instanceDataRetriever also to make this easier to run locally with just a different implementation.

Comment (zmarois) [reply priam/src/main/java/com/netflix/priam/defaultimpl/PriamConfiguration.java:676]: Good catch, I messed up here.

My primary purpose of doing this is just to defer the call to getPublicHostname/getPublicIP to the time it is actually used. My instances in a private VPC don't have these, so I didn't personally run into an issue.

Would you like this upstream if done correctly? I'd like to search for other usages of instance metadata and isolate into the instanceDataRetriever also to make this easier to run locally with just a different implementation.

Comment (arunagrawal84) [reply priam/src/main/java/com/netflix/priam/defaultimpl/PriamConfiguration.java:676]: Yes. I would like this in upstream to make code cleaner and later as you said to run locally. One of the goal for 2018 is to ensure we can run Priam locally for testing.

**Pull request merged**